components/pages/eventtypes/EventTypeList.tsx

```diff
@@ -7,25 +7,17 @@ import React, { Fragment } from "react";
 import classNames from "@lib/classNames";
 import { useLocale } from "@lib/hooks/useLocale";
 import showToast from "@lib/notification";
+import { inferQueryOutput } from "@lib/trpc";
 
 import { Tooltip } from "@components/Tooltip";
 import EventTypeDescription from "@components/eventtype/EventTypeDescription";
 import AvatarGroup from "@components/ui/AvatarGroup";
 
+type EventTypes = inferQueryOutput<"viewer.eventTypes">["eventTypes"];
 interface Props {
-  profile: { slug: string };
+  profile: { slug: string | null };
   readOnly: boolean;
-  types: {
-    $disabled: boolean;
-    hidden: boolean;
-    id: number;
-    slug: string;
-    title: string;
-    users: {
-      name: string;
-      avatar: string;
-    }[];
-  };
+  types: EventTypes[number]["eventTypes"];
```

Comment:
Easier to read if you split up the logic a bit

```suggestion
type EventTypeGroup = inferQueryOutput<"viewer.eventTypes">["eventTypes"][number];
type EventType = EventTypeGroup["eventTypes"][number];
interface Props {
  profile: { slug: string | null };
  readOnly: boolean;
  types: EventType[];
}
```

Also when doing this you see that what you return on `<"viewer.eventTypes">["eventTypes"]` should probably be called `<"viewer.eventTypes">["eventTypeGroups"]`

Reply:
Also, to echo what I wrote earlier - this is clearly tied to the `viewer.eventTypes` so this component should be inlined in whatever file actually does the `viewer.eventTypes`-useQuery